exercises/minesweeper/test/test_minesweeper.c

```diff
@@ -0,0 +1,171 @@
+#include "vendor/unity.h"
+#include "../src/minesweeper.h"
+#include <stdlib.h>
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static void free_all(char **field, const size_t rows)
+{
+   for (size_t i = 0; i < rows; i++)
+      free(field[i]);
+
+   free(field);
+}
+
+void test_annotate_no_rows(void)
+{
+   const char **minefield = NULL;
+   const size_t rows = 0;
+   char **actual = annotate(minefield, rows);
+   TEST_ASSERT_NULL(actual);
+}
+
+void test_annotate_no_columns(void)
+{
+   TEST_IGNORE();               // delete this line to run test
+   const char *minefield[] = { "" };
+   const char *expected[] = { "" };
+   const size_t rows = ARRAY_SIZE(expected);
```

Comment:
Just spotted this problem with using the array size macro in the test file (may apply to previous exercises also!): The macro is defined in the implementation header yet is not provided to the student. Generally I would think that things not directly related to the implementation (e.g. enums) that are used in the test file should be defined in the test file and let the student be concerned with the implementation rather than the instrumentation. This particular macro is used in both implementation and test, but only by coincidence for _this particular implementation_, students OTOH may find other implementations that do not require such a macro. To resolve this the macro should be defined in the test and separately in the implementation's source file.

Reply:
In fact, I just noticed, in this implementation the macro is not used at all in the source and so doesn't need to be defined there at all, just in the test file

Reply:
Fully agree, will check and fix them all.